checker/src/org/checkerframework/checker/oigj/OwnershipAnnotatedTypeFactory.java

```diff
@@ -47,10 +53,17 @@ public OwnershipAnnotatedTypeFactory(BaseTypeChecker checker) {
 
     @Override
     protected TreeAnnotator createTreeAnnotator() {
-        return new ListTreeAnnotator(
-                super.createTreeAnnotator(),
-                new OwnershipTreeAnnotator(this)
-        );
+        ImplicitsTreeAnnotator implicitsTreeAnnotator = new ImplicitsTreeAnnotator(this);
+        implicitsTreeAnnotator.addTreeKind(Tree.Kind.NEW_CLASS, MUTABLE);
+
+        implicitsTreeAnnotator.addTreeKind(Tree.Kind.CLASS, BOTTOM_QUAL);
+        implicitsTreeAnnotator.addTreeKind(Tree.Kind.ENUM, BOTTOM_QUAL);
+        implicitsTreeAnnotator.addTreeKind(Tree.Kind.ANNOTATION_TYPE, BOTTOM_QUAL);
+        implicitsTreeAnnotator.addTreeKind(Tree.Kind.NEW_ARRAY, BOTTOM_QUAL);
+
+
+        return new ListTreeAnnotator(implicitsTreeAnnotator, new PropagationTreeAnnotator(this),
```

Comment:
It's not nice that there is no call of the super implementation anymore and instead you had to copy the current implementation here.
If at some point we change super.createTreeAnnotator(), this implementation will not benefit from the change.